migrationtools/migrations.py

```diff
@@ -253,6 +297,243 @@ def migrate_registrations(self, start=None, stop=None, limit=None):
             self.echo(" Completed")
         self.echo('Registation Migration completed')
 
+    def migrate_subscriptions(self, active_only=True, start=None, stop=None, limit=None):
+        # Use these shortcuts to make the code a bit more readable.
+        sub_cols = self.ndoh_control.subscription.c
+        ms_cols = self.ndoh_control.messageset.c
+        ident_cols = self.seed_identity.identity.c
+        sched_cols = self.seed_sbm.schedule.c
+
+        migration_id = str(uuid.uuid4())
+        migration_type = 'registration'
+        self.echo('Starting registration migration ({0})'.format(migration_id))
+        for subscription in self.ndoh_control.get_subscriptions(active_only, start, stop, limit=limit):
+            # Keep a record of all transactions started so they can all be rolled back on any error.
+            transactions = {}
+
+            sub_id = subscription[sub_cols.id]
+            self.echo("Starting migration of {id} ...".format(id=sub_id), nl=False)
+
+            # Get basic details from the sub record
+            msisdn = subscription[sub_cols.to_addr]
+            lang = subscription[sub_cols.lang]
+            old_messageset_name = subscription[ms_cols.short_name]
+            created_at = subscription[sub_cols.created_at]
+            updated_at = subscription[sub_cols.updated_at]
+            active = subscription[sub_cols.active]
+
+            # Lookup if there is an existing Seed Identity for this msisdn.
+            identity = self.seed_identity.lookup_identity_with_msdisdn(msisdn)
+
+            # Setup a seed identity transaction:
+            transactions['seed_identity'] = self.seed_identity.start_transaction()
+
+            if identity is None:
+                ident_lang = None
+                # Lookup Vumi Contact info for this msisdn.
+                vumi_id = subscription[sub_cols.contact_key]
+                vumi_contact = self.vumi_contacts.lookup_contact(vumi_id)
+
+                if vumi_contact is None:
+                    consent = False
+                    sa_id_no = None
+                    mom_dob = None
+                    source = None
+                    vumi_lang = None
+                else:
+                    consent = vumi_contact['json']['extra'].get('consent', False)
+                    sa_id_no = vumi_contact['json']['extra'].get('sa_id', False)
+                    mom_dob = vumi_contact['json']['extra'].get('dob', False)
+                    vumi_lang = vumi_contact['json']['extra'].get('language_choice', None)
```

Comment:
Do you still want me to convert the strings to JSON types in the db dump? This hints at you already having done that?